middleware/grpc_logging_test.go

```diff
@@ -0,0 +1,30 @@
+package middleware
+
+import (
+	"context"
+	"testing"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"google.golang.org/grpc"
+
+	"github.com/weaveworks/common/logging"
+)
+
+func BenchmarkGRPCServerLog_UnaryServerInterceptor_NoError(b *testing.B) {
```

Comment:
Could you share benchmark results? How to run this benchmark:

1. Run the benchmark both with `master` and this PR code, storing each result to file (I've written the command without testing it, hope I didn't introduce any typo):
   ```
   go test -run '^$' -bench 'BenchmarkGRPCServerLog_UnaryServerInterceptor_NoError' -benchmem ./middleware > before.txt
   ```
2. Use [`benchstat`](https://pkg.go.dev/golang.org/x/perf/cmd/benchstat) to compare results:
   ```
   benchstat before.txt after.txt
   ```

Reply:
```
name                                             old time/op    new time/op    delta
GRPCServerLog_UnaryServerInterceptor_NoError-16     890ns ± 0%      31ns ± 0%   ~     (p=1.000 n=1+1)

name                                             old alloc/op   new alloc/op   delta
GRPCServerLog_UnaryServerInterceptor_NoError-16    1.12kB ± 0%    0.00kB        ~     (p=1.000 n=1+1)

name                                             old allocs/op  new allocs/op  delta
GRPCServerLog_UnaryServerInterceptor_NoError-16      16.0 ± 0%       0.0        ~     (p=1.000 n=1+1)
```

Reply:
Sorry, I forgot one important detail. TO get `benchstat` comparison working, you need to run the benchmark at least twice. Run the test as above, but add `-count=3`